src/widget/wtracktableview.cpp

```diff
@@ -946,7 +946,7 @@ void WTrackTableView::applySorting() {
         return;
     }
 
-    Qt::SortOrder sortOrder = m_pSortOrder->get() ? Qt::DescendingOrder : Qt::AscendingOrder;
+    Qt::SortOrder sortOrder = (m_pSortOrder->get() == 0) ? Qt::AscendingOrder : Qt::DescendingOrder;
```

Comment:
toBool()?

Reply:
That would work, but IMHO it's semantically correct. The sort order is an enumeration with two values, not a boolean.